<launch>

  <!--  **************** Synchronization ****************  -->
  <param name="/use_sim_time" value="true"/>

  <!-- ***************** YAML file path *****************-->
  <arg name="map_file"       default="$(find Project_2)/maps/grid_mapping.yaml"/>

  <!--  ****** AMCL *****  -->
  <include file="$(find Project_2)/launch/amcl.launch.xml"/>

	<!--  **************** Visualization ****************  -->
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find Project_2)/rviz/Project2.rviz"/>

  <!--  **************** Drawing ****************  -->
  <node pkg="Project_2" type="draw.py" name="draw" />

	<!--  **************** Odometry ****************  -->
	<node pkg="Project_2" type="pub_odom" name="pub_odom"/>

  <!-- ***************** Smoother map ***************** -->
  <node pkg="Project_2" type="occupancy_grid_processing.py" name="occupancy_grid_processing" output="log"/>
  
  <!--  **************** Saving ****************  -->
	<node pkg="Project_2" type="save_map.py" name="save_map" output="log"/>

  <!--  **************** Map Server ****************  -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
        <param name="frame_id" value="map"/>
    </node>
  
</launch>
